central/signatureintegration/datastore/singleton.go

```diff
@@ -1,21 +1,58 @@
 package datastore
 
 import (
+	"context"
+
+	"github.com/pkg/errors"
 	"github.com/stackrox/rox/central/globaldb"
 	policyDataStore "github.com/stackrox/rox/central/policy/datastore"
+	"github.com/stackrox/rox/central/signatureintegration/store"
 	pgStore "github.com/stackrox/rox/central/signatureintegration/store/postgres"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/features"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/signatures"
 	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/utils"
 )
 
 var (
 	once     sync.Once
 	instance DataStore
 )
 
+// setupDefaultRedHatSignatureIntegration ensures the presence of the default Red Hat signature integration is
+// in line with the RedHatImagesSignedPolicy feature flag.
+func setupDefaultRedHatSignatureIntegration(siStore store.SignatureIntegrationStore) {
+	log.Debugf("Setting up default Red Hat signature integration %q (%s)",
```

Comment:
This log may be confusing if the feature is turned off. Perhaps log it later or not at all

Reply:
Good catch, thanks! 100% agreed. I have made some changes to hopefully improve this: https://github.com/stackrox/stackrox/pull/15761/commits/7349ff8fd315a83fc3cb74b0ab76d68ff5097b6c

Reply:
This is also outdated now, since the flag-handling logic has been removed: https://github.com/stackrox/stackrox/pull/15761/commits/14ebb647a62e0ea63dac468a2f620f68f2d7ff04 (see commit msg for details on why)